rules/python/third_parties/aws_query_injection.yml

```diff
@@ -0,0 +1,143 @@
+imports:
+  - python_shared_common_user_input
+  - python_shared_lang_import1
+patterns:
+  # simple db
+  - pattern: $<CALLER>.select($<...>SelectExpression=$<USER_INPUT>$<...>)
+    filters:
+      - variable: CALLER
+        detection: python_third_parties_aws_query_injection_simpledb_caller
+        scope: result
+      - variable: USER_INPUT
+        detection: python_third_parties_aws_query_injection_user_input
+        scope: result
+  # dynamodb
+  - pattern: $<CALLER>.$<METHOD>($<...>FilterExpression=$<USER_INPUT>$<...>)
+    filters:
+      - variable: CALLER
+        detection: python_third_parties_aws_query_injection_dynamodb_caller
+        scope: result
+      - variable: METHOD
+        values:
+          - query
+          - scan
+      - variable: USER_INPUT
+        detection: python_third_parties_aws_query_injection_user_input
+        scope: result
+  - pattern: $<CALLER>.update_item($<...>UpdateExpression=$<USER_INPUT>$<...>)
+    filters:
+      - variable: CALLER
+        detection: python_third_parties_aws_query_injection_dynamodb_caller
+        scope: result
+      - variable: USER_INPUT
+        detection: python_third_parties_aws_query_injection_user_input
+        scope: result
+  # legacy dynamodb
+  - pattern: $<CALLER>.query($<...>QueryFilter=$<USER_INPUT>$<...>)
+    filters:
+      - variable: CALLER
+        detection: python_third_parties_aws_query_injection_dynamodb_caller
+        scope: result
+      - variable: USER_INPUT
+        detection: python_third_parties_aws_query_injection_user_input
+        scope: result
+  - pattern: $<CALLER>.scan($<...>ScanFilter=$<USER_INPUT>$<...>)
+    filters:
+      - variable: CALLER
+        detection: python_third_parties_aws_query_injection_dynamodb_caller
+        scope: result
+      - variable: USER_INPUT
+        detection: python_third_parties_aws_query_injection_user_input
+        scope: result
+auxiliary:
+  - id: python_third_parties_aws_query_injection_simpledb_caller
+    patterns:
+      - pattern: $<BOTO3_CLIENT>('sdb'$<...>)
+        filters:
+          - variable: BOTO3_CLIENT
+            detection: python_shared_lang_import1
+            scope: cursor
+            filters:
+              - variable: MODULE1
+                values: [boto3]
+              - variable: NAME
+                values: [client]
+  - id: python_third_parties_aws_query_injection_dynamodb_caller
+    patterns:
+      - pattern: $<DYNAMODB_INIT>.Table($<...>)
+        filters:
+          - variable: DYNAMODB_INIT
+            detection: python_third_parties_aws_query_injection_dynamodb_init
+            scope: result
+      - pattern: $<BOTO3_CLIENT>('dynamodb'$<...>)
```

Comment:
This probably isn't very likely but if you use a `string_regex` filter for the `'dynamodb'` part then it would work when that's a constant/variable.

Reply:
Good shout 🙏